src/runner.rs

```diff
@@ -52,20 +57,46 @@ pub trait Runner {
     fn end(&mut self, _spawn: &mut Self::Spawn) {}
 }
 
-/// Allows spawn a task to the thread pool from a different thread.
+/// Allows spawning a task to the thread pool from a different thread.
 pub trait RemoteSpawn: Sync + Send {
     type Task;
+    type SpawnOption;
```

Comment:
Why choose "Option" for `Remote` and "Context" for `Local`?

Reply:
I wanted to express `Option` is something that is usually controlled by a user while `Context` is not set by the user but comes from inside the queue in most cases.